versioned_hdf5/api.py

```diff
@@ -67,6 +68,17 @@ def __init__(self, f):
         self.f = f
         if '_version_data' not in f:
             initialize(f)
+        else:
+            # This is not a new file; check data version identifier for compatibility
+            if self.data_version_identifier != DATA_VERSION:
+                warnings.warn(
```

Comment:
I think that's the right idea, but unfortunately we have 100,000s of old files which would then trigger those warnings. Can we make them suppressed by default?

Reply:
Would a log message be better? Maybe emitted at `INFO` level, so it's suppressed by default?

Reply:
I made this an INFO level log message here.